perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3071,7 +3071,7 @@ private void updateExistingMembersWhileSynchronization(PerunSession sess, Group
 									perunBl.getMembersManagerBl().expireMember(sess, richMember);
 									log.info("Switching member id {} to EXPIRE state, due to expiration {}.", richMember.getId(), membershipExpiration.getValue());
 									log.debug("Switching member to EXPIRE state, additional info: membership expiration date='{}', system now date='{}'", currentMembershipExpirationDate, now);
-								} catch (MemberNotValidYetException e) {
+								} catch (MemberNotValidYetException | WrongReferenceAttributeValueException | WrongAttributeValueException e) {
```

Comment:
There is a minor problem outside the scope of your change. If the expiration of member fails on attributes, then he is kept in disabled state. Later in this method it tries to validate any still disabled member. Hence we call validation twice for no reason and second one will switch him to valid (disregarding expiration date). We should merge / rework following "IFs" into single one, so that only one branch is used on member.

Reply:
This par of code was reworked.